linker/shared/src/main/scala/org/scalajs/linker/backend/emitter/VarGen.scala

```diff
@@ -208,11 +208,23 @@ private[emitter] final class VarGen(jsGen: JSGen, nameGen: NameGen,
   def typeRefVar(field: String, typeRef: NonArrayTypeRef)(
       implicit moduleContext: ModuleContext, globalKnowledge: GlobalKnowledge,
       pos: Position): Tree = {
+    /* Explicitly bringing `PrimRefScope` and `ClassScope` as local implicit
+     * vals should not be necessary, and used not to be there. When upgrading
+     * to sbt 1.5.4 from 1.2.8, compilation started failing because of missing
+     * implicit `Scope[PrimRef]` and `Scope[ClassName]` in this method, in
+     * *some* environments (depending on machine, OS, JDK version, in or out
+     * IDE, regular compile versus Scaladoc, etc., perhaps the phase of the
+     * moon, for all I could tell). It is not even clear that the sbt version
+     * is actually relevant.
+     * Regardless, the explicit vals reliably fix the issue.
+     */
     typeRef match {
       case primRef: PrimRef =>
+        implicit val primRefScope: Scope[PrimRef] = Scope.PrimRefScope
         globalVar(field, primRef)
 
       case ClassRef(className) =>
+        implicit val classScope: Scope[ClassName] = Scope.ClassScope
         globalVar(field, className)
     }
   }
```

Comment:
@gzm0 I had to add the above changes to make the CI pass. There were errors like
```
[error] linker/shared/src/main/scala/org/scalajs/linker/backend/emitter/VarGen.scala:213:18: could not find implicit value for evidence parameter of type VarGen.this.Scope[org.scalajs.ir.Types.PrimRef]
[error] Error occurred in an application involving default arguments.
[error]         globalVar(field, primRef)
[error]                  ^
[error] linker/shared/src/main/scala/org/scalajs/linker/backend/emitter/VarGen.scala:216:18: could not find implicit value for evidence parameter of type VarGen.this.Scope[org.scalajs.ir.Names.ClassName]
[error] Error occurred in an application involving default arguments.
[error]         globalVar(field, className)
[error]                  ^
[error] two errors found
[error] Scaladoc generation failed
```
When trying to investigate, I also observed the same error on my local Windows machine, but only in the IDE. That was without regenerating the build, so supposedly the IDE was still using the config generated through sbt 1.2.8.

Is that OK for you?

Reply:
If it's the only thing we can do :-/ In the past for me, the incremental compiler failed on this specific thing if there were other errors in the same file. IIRC, cleaning helped.

Are we 100% sure the CI environment is clean?

Reply:
We do `git clean -fdx && rm -rf partest/fetchedSources/` at the beginning of each CI job, so yes, it's pretty clear that it is clean.

Reply:
Also `linker$v/compile` succeeds, during the same CI run, before `linker$v/compile:doc` fails, so ...